src/figuresClasses.ts

```diff
@@ -1,11 +1,74 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('throws an error');
+    } else if (a + b <= c || c + b <= a || c + a <= b) {
+      throw new Error(`
+        throws an error: sides ${a}, ${b} and ${c} can’t form a triangle
+      `);
+    }
+  }
+
+  getArea(): number {
```

Comment:
The 'getArea' function should not accept any parameters, as all necessary values are already available as object fields.